app/components/FilterableComponents/FilterableTableLayout.tsx

```diff
@@ -1,29 +1,81 @@
-import React from 'react';
+import React, {useMemo} from 'react';
 import {Col, Table, Alert} from 'react-bootstrap';
 import SortableTableHeader from './SortableTableHeader';
 import FilterableTableHeaders from './FilterableTableHeaders';
-import {ISearchProps} from 'components/Search/SearchProps';
+import {ISearchProps, ISearchExtraFilter} from 'components/Search/SearchProps';
+import {useRouter} from 'next/router';
+import safeJsonParse from 'lib/safeJsonParse';
+import {FilterArgs} from 'components/Search/ISearchOption';
 
 interface Props extends ISearchProps {
   body: JSX.Element;
   isLoading?: boolean;
-  extraControls?: JSX.Element;
+  extraFilters?: ISearchExtraFilter[];
 }
-export const FilterableTableLayoutComponent: React.FunctionComponent<Props> = (
-  props
-) => {
-  const {searchOptions, body, isLoading} = props;
+export const FilterableTableLayoutComponent: React.FunctionComponent<Props> = ({
+  searchOptions,
+  body,
+  isLoading,
+  extraFilters = []
+}) => {
+  const router = useRouter();
+  const relayVars = useMemo(
+    () => safeJsonParse(router.query.relayVars as string),
+    [router]
+  );
+
+  const applySearch = (searchData: FilterArgs) => {
+    const newQuery = {
+      // copy the vars from the query string, so that the args coming from extraFilters are not overriden
+      ...relayVars,
+      ...searchData
+    };
+    searchOptions.forEach((option) => {
+      const column = option.columnName;
+
+      if (option.isSearchEnabled) {
+        newQuery[column] = searchData[column] ?? undefined;
+      }
+    });
+
+    const queryString = JSON.stringify(newQuery);
+
+    const url = {
+      pathname: router.pathname,
+      query: {
+        ...router.query,
+        relayVars: queryString,
+        pageVars: JSON.stringify({offset: 0})
```

Comment:
This is where we lose the pagination 'result per page' setting on reset. It is also breaking SRP, the filtering should not know about pagination specifics

Reply:
I am planning to move the page change handling in this component (so that all interactions with the router are constrained to the one file)

Reply:
Should we break it in 2 pieces then?
- a small non-UI, pure TS, router interaction service
- the `FilterableTableLayout` that deals with react rendering